rotkehlchen/tests/utils/blockchain.py

```diff
@@ -84,7 +85,10 @@ def assert_eth_balances_result(
     if not totals_only:
         per_account = result['per_account']
         if also_btc:
-            assert len(per_account) == 2
+            if btc_queried is True:
+                assert len(per_account) == 2
+            else:
```

Comment:
What are you doing here? No no no .... undo all changes to this function. This testing function should not change. There is no reason to touch this

Reply:
The reason why I touched this is because previously `BalancesUpdate` was returned after account addition and it contained all the known balances (eth, btc, etc.) and now it is not returned. So the only way now to get all balances (including btc) is to query all balances (we don't get this information from a blockchain-specific request). But querying all balances cannot be done in some tests, so I don't understand how we can avoid the modification of this function.

Reply:
you would have to modify test to add balance queries where this function was used. This is what I told you 2 days ago: https://github.com/rotki/rotki/pull/4667#pullrequestreview-1067918956

> Otherwise you have to modify the test to also add balance query api calls and see they work fine.